tokens/src/main/java/com/telefonica/mistica/tokens/compose/GenerateBrandColors.kt

```diff
@@ -0,0 +1,111 @@
+package com.telefonica.mistica.tokens.compose
+
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.KModifier
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.TypeSpec
+import com.telefonica.mistica.tokens.TokensGenerator
+import com.telefonica.mistica.tokens.common.GetColorNameWithAlpha
+import com.telefonica.mistica.tokens.common.GetColorsWithAlpha
+import com.telefonica.mistica.tokens.compose.GenerateComposeFiles.Companion.colorClass
+import com.telefonica.mistica.tokens.compose.GenerateComposeFiles.Companion.misticaColorsClass
+import com.telefonica.mistica.tokens.dto.ColorDTO
+import com.telefonica.mistica.tokens.dto.TokensDTO
+import com.telefonica.mistica.tokens.xml.GenerateXMLFiles.Companion.capitalizeString
+import java.io.File
+
+/**
+ * Generates the XXXBrandColorsAutogenerated.kt file for each brand.
+ */
+class GenerateBrandColors(
+    private val getColorsWithAlpha: GetColorsWithAlpha = GetColorsWithAlpha(),
+    private val getColorNameWithAlpha: GetColorNameWithAlpha = GetColorNameWithAlpha(),
+) {
+
+    operator fun invoke(tokens: TokensDTO, brand: String) {
+        val paletteClassName = "${brand.capitalizeString()}PaletteColor"
+
+        val lightProperty = PropertySpec.builder("lightColors", misticaColorsClass)
+            .initializer(getColorsConstructor(brand, tokens.light, paletteClassName))
+            .build()
+
+        val darkProperty = PropertySpec.builder("darkColors", misticaColorsClass)
+            .initializer(getColorsConstructor(brand, tokens.dark, paletteClassName))
+            .build()
+
+        val colorsObject = TypeSpec.objectBuilder("${brand.capitalizeString()}BrandColors")
+            .addProperty(lightProperty)
+            .addProperty(darkProperty)
+            .build()
+
+        val paletteObject = TypeSpec.objectBuilder(paletteClassName)
+            .addModifiers(KModifier.PRIVATE)
+            .addProperties(getPaletteProperties(tokens, brand))
+            .build()
+
+        val file = FileSpec.builder("com.telefonica.mistica.compose.theme.brand", "${brand.capitalizeString()}BrandColorsAutogenerated")
+            .addFileComment(TokensGenerator.AUTOGENERATED_COMMENT)
+            .addType(colorsObject)
+            .addType(paletteObject)
+            .build()
+
+        file.writeTo(File(GenerateComposeFiles.LIBRARY_CODE_PATH))
+    }
+
+    private fun getColorsConstructor(
+        brand: String,
+        colors: Map<String, ColorDTO>,
+        paletteClassName: String,
+    ): String {
+
+        var colorsConstructor = "${GenerateComposeFiles.MISTICA_COLORS}("
+
+        colors.forEach { color ->
```

Comment:
Consider to use destructuring declaration in that cases in order to avoid misunderstanding between color item and ColorDTO object and make it more readable.
```suggestion
        colors.forEach { (key, colorDto) ->
```
E.g. changing "color.value.value" to "colorDto.value"